### PR TITLE
[Task-22][API] As a user, I can get my keywords & [Task-23][API] As a user, I can keyword search detail

### LINKED ISSUE #48: [Task-22][API] As a user, I can get my keywords
## Why

User should get an API to get keyword list

## Acceptance Criteria

Test using postman and test codes

## Who Benefits?

Users

### LINKED ISSUE #49: [Task-23][API] As a user, I can keyword search detail
## Why

User can get keyword search detail using API

## Acceptance Criteria

Test using postman and test code

## Who Benefits?

Users

### PR DESCRIPTION
**Pull request by rafayet-monon**
Task Card 
- https://github.com/rafayet-monon/elixir_search_extractor/projects/1#card-64840269
- https://github.com/rafayet-monon/elixir_search_extractor/projects/1#card-64840588

## What happened 👀

API for keyword list and detail

## Insight 📝

- [x] Keyword list API
- [x] Keyword detail API 

## Proof Of Work 📹

- Keyword List
image

- Keyword detail success
image

- Keyword detail failed
image

### CODE REVIEW

File: test/elixir_search_extractor/search_keyword/search_keywords_test.exs [hunk 1]
```diff
@@ -48,6 +48,24 @@ defmodule ElixirSearchExtractor.SearchKeyword.SearchKeywordsTest do
...
+      user = user_fixture()
+      user_file = insert(:keyword_file, user: user)
+      user_keyword = insert(:keyword, keyword_file: user_file)
+
+      retrieved_keyword = SearchKeywords.get_user_keyword(user, user_keyword.id)
+
+      assert retrieved_keyword.id == user_keyword.id
+    end
+
+    test "returns nil if the user keyword does not exists" do
```
Comment (hanam1ni) [inline test/elixir_search_extractor/search_keyword/search_keywords_test.exs]:
```suggestion
    test "returns nil if the user keyword does not exist" do
```

File: test/elixir_search_extractor/search_keyword/search_keywords_test.exs [hunk 2]
```diff
@@ -48,6 +48,24 @@ defmodule ElixirSearchExtractor.SearchKeyword.SearchKeywordsTest do
     end
   end
 
+  describe "get_user_keyword/2" do
```
Comment (hanam1ni) [inline test/elixir_search_extractor/search_keyword/search_keywords_test.exs:51]: Let's add another test case when getting a keyword that belongs to the other user.

File: lib/elixir_search_extractor_web/controllers/api/v1/keyword_controller.ex [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    conn
+    |> put_view(ListView)
+    |> render("index.json", %{data: keywords, meta: meta_data(pagination)})
+  end
+
+  def show(conn, %{"id" => keyword_id}) do
+    case SearchKeywords.get_user_keyword(conn.assigns.current_user, keyword_id) do
+      nil ->
+        conn
+        |> put_status(:unprocessable_entity)
```
Comment (hanam1ni) [inline lib/elixir_search_extractor_web/controllers/api/v1/keyword_controller.ex]: Should this return `not_found` status instead?

```suggestion
        |> put_status(:not_found)
```

**Comment by rafayet-monon** [comment]
@hanam1ni Fixed reviews in [5b99cfe](https://github.com/rafayet-monon/elixir_search_extractor/pull/55/commits/5b99cfe269737f80544efc5733ad4ce249e21328)

**Review by hanam1ni — APPROVED** [review]

File: test/elixir_search_extractor/search_keyword/search_keywords_test.exs [hunk 4]
```diff
@@ -34,20 +34,47 @@ defmodule ElixirSearchExtractor.SearchKeyword.SearchKeywordsTest do
   end
 
   describe "get_keyword!/1" do
-    test "returns the keyword if the keyword exists" do
+    test "returns the keyword if the keyword exist" do
```
Comment (rosle) [inline test/elixir_search_extractor/search_keyword/search_keywords_test.exs]:
```suggestion
    test "returns the keyword if the keyword exists" do
```

File: test/elixir_search_extractor/search_keyword/search_keywords_test.exs [hunk 5]
```diff
@@ -34,20 +34,47 @@ defmodule ElixirSearchExtractor.SearchKeyword.SearchKeywordsTest do
...
-    test "raises Ecto.NoResultsError if the keyword does not exists" do
+    test "raises Ecto.NoResultsError if the keyword does not exist" do
       assert_raise Ecto.NoResultsError, fn ->
         SearchKeywords.get_keyword!(1)
       end
     end
   end
 
+  describe "get_user_keyword/2" do
+    test "returns the user keyword if the keyword exist" do
```
Comment (rosle) [inline test/elixir_search_extractor/search_keyword/search_keywords_test.exs]:
```suggestion
    test "returns the user keyword if the keyword exists" do
```

**Review by rosle — APPROVED** [review]

**Pull request merged**